va/va_test.go

```diff
@@ -257,8 +257,9 @@ func (inmem inMemVA) IsCAAValid(ctx context.Context, req *vapb.IsCAAValidRequest
 func TestValidateMalformedChallenge(t *testing.T) {
 	va, _ := setup(nil, 0, "", nil, nil)
 
-	_, prob := va.validateChallenge(ctx, dnsi("example.com"), createChallenge("fake-type-01"))
+	_, err := va.validateChallenge(ctx, dnsi("example.com"), createChallenge("fake-type-01"))
 
+	prob := detailedError(err)
```

Comment:
So this is actually what I originally did, and I discovered a problem with it: for some errors, the `validateHTTP` / `validateDNS` / etc functions assign a particular error code. But for most errors, they are simply passed through (e.g. a `net.OpError`) and don't get assigned their error code and humanified until `detailedError` is called.

So I went back through and changed (nearly) everything to call `detailedError` first. The call sites you see where I used `AssertErrorIs` are some of the ones where the error code is assigned by some function other than `detailedError`.

I honestly don't love that the tests rely on calling a second private function in order to interpret the output of the functions under test, but it seemed like the best way that I could think of, since what we most want to test is the user-facing output, that is to say the ProblemDetails.